src/window.rs

```diff
@@ -0,0 +1,149 @@
+//! Displays an image in a window created by sdl2.
+
+use image::RgbaImage;
+use sdl2::event::{WindowEvent, Event};
+use sdl2::rect::Rect;
+use sdl2::keyboard::Keycode;
+use sdl2::pixels::{Color, PixelFormatEnum};
+use sdl2::surface::Surface;
+
+/// Displays the provided RGBA image in a new window.
+pub fn display_image(title: &str, image: &RgbaImage, window_width: u32, window_height: u32) {
+
+    let image_width = image.width();
+    let image_height = image.height();
+
+    // resizes and returns the image that will be used to display in the window
+    let (output_image_width, output_image_height, output_image) = 
+        if image_height < window_height && image_width < window_width {
+            (image_width, image_height, image.clone())
+        } else {
+            // scale is used to determine how small an image has to be resized to fit within
+            // the provided window dimensions
+            let scale = {
+                let width_scale = window_width as f32 / image_width as f32; 
+                let height_scale = window_height as f32 / image_height as f32;
+                if  width_scale < height_scale {
+                    width_scale
+                } else {
+                    height_scale
+                }
+            };
+            let height = (scale * image_height as f32) as u32;
+            let width = (scale * image_width as f32) as u32;
+            let output_image = image::imageops::resize(image, width, height, image::FilterType::Triangle);
+            (width, height, output_image)
+    };
+
+    const CHANNEL_COUNT: u32 = 4;
+    let pitch  = output_image_width * CHANNEL_COUNT;
```

Comment:
Nit: two spaces after "pitch" - worth running rustfmt and using whatever formatting that creates?